libs/core/src/lib/status-indicator/status-indicator.component.ts

```diff
@@ -0,0 +1,363 @@
+export type FillingType = 'radial' | 'angled' | 'linearup' | 'lineardown' | 'linearleft';
+export type FillingDirection = 'clock' | 'anticlock';
+
+export interface Points {
+    x: number;
+    y: number;
+}
+
+@Component({
+    selector: 'fd-status-indicator',
+    templateUrl: './status-indicator.component.html',
+    styleUrls: ['./status-indicator.component.scss'],
+    host: {
+        '[attr.aria-label]': 'ariaLabel',
+        '[attr.aria-roledescription]': 'ariaRoledescription',
+        '[attr.focusable]': 'focusable',
+        '[attr.title]': 'title',
+        '[attr.role]': 'role',
+        '[attr.aria-valuetext]': 'ariaValuetext'
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class StatusIndicatorComponent implements OnChanges, OnInit, AfterViewInit {
+    /**
+     * value id defines the id of the object.
+     */
+    @Input()
+    id: string;
+
+    /**
+     * defines the size of the status indicator.
+     * Can be one of the following: 'sm' | 'md' | 'lg' | 'xl'
+     * Default size is Medium(md).
+     */
+    @Input()
+    size: Size = 'md';
+
+    /**
+     * The status represented by the Status Indicator.
+     * Can be one of the following: 'negative' | 'critical' | 'informative'
+     * For default Object Status omit this property
+     */
+    @Input()
+    status: StatusIndicatorColor;
+
+    /** Define the text content of the Status indicator*/
+    @Input()
+    statusLabel: string;
+
+    /**
+     * positioning of the status indicator image withing the defined height and width .
+     */
+    @Input()
+    viewBox: string;
+
+    /**
+     * boolean value to define requirement of the label.
+     */
+    @Input()
+    hasLabel: boolean;
+
+    /**
+     * boolean value to be marked as a clickable
+     */
+    @Input()
+    clickable: boolean;
+
+    /** Aria label for the Status Indicator. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Aria defines role description for the Status Indicator. */
+    @Input()
+    ariaRoledescription: string = null;
+
+    /** Aria Focusable for the Status Indicator. */
+    @Input()
+    focusable: boolean;
+
+    /** Aria Role for the Status Indicator. */
+    @Input()
+    role: string = null;
+
+    /** Aria Value Text for the Status Indicator. */
+    @Input()
+    ariaValuetext: string = null;
+
+    /** Aria title for the status indicator. */
+    @Input()
+    title: string = null;
+
+    /** defines the label position the value can be 'left' | 'right' | 'top' | 'bottom' */
+    @Input()
+    labelPosition: LablePosition;
+
+    @Input()
+    path: string[];
+
+    /**
+     * Offset value to be filled under the give percentatge value.
+     */
+    @Input()
+    fillPercentage: number;
+
+    /**
+     * value to define fill direction
+     */
+    @Input()
+    fillDirection: FillingDirection = 'clock';
+
+    get FillDirection(): FillingDirection {
```

Comment:
please use camelCase notation
```suggestion
    get _fillDirection(): FillingDirection {
```